### PR TITLE
Update workload identity docs

### PR DESCRIPTION
**Pull request by melugoyal**
adding some changes that were lost when https://github.com/astronomer/docs/pull/2271 was closed. specifically this clears up some confusion in the AWS workload identity docs where the cluster-level IAM role was still being referenced, and fixes an inaccuracy in the docs about leveraging the AWS airflow connection.

### CODE REVIEW

File: astro/connect-aws.md [hunk 1]
```diff
@@ -173,20 +173,19 @@ To grant an Astro cluster access to a service that is running in an AWS account
...
                     ]
                 },
                 "Action": "sts:AssumeRole"
             }
         ]
     }
     ```
-    
-    The Astro cluster includes the `AirflowLogsS3-<clusterid>` role. When you configure an [AWS Airflow Connection](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/connections/aws.html) for a Deployment, use `"arn:aws:iam::<dataplane-AWS-account-ID>:role/AirflowS3Logs-<cluster-ID>"` as the value for `aws_arn`.
+    When you configure an [AWS Airflow Connection](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/connections/aws.html) for a Deployment, use the Workload Identity value as the value for `aws_arn`.
```
Comment (melugoyal) [inline astro/connect-aws.md:184]: @chomipi88 should we just remove this line now?

File: astro/connect-aws.md [hunk 2]
```diff
@@ -156,7 +156,7 @@ This connection option is only available for dedicated Astro Hosted clusters and
 
 :::
 
-To grant an Astro cluster access to a service that is running in an AWS account not managed by Astronomer, use AWS IAM roles. IAM roles on AWS are often used to manage the level of access a specific user, object, or group of users has to a resource. This includes an Amazon S3 bucket, Redshift instance, or secrets backend.
+To grant an Astro deployment access to a service that is running in an AWS account not managed by Astronomer, use AWS IAM roles. IAM roles on AWS are often used to manage the level of access a specific user, object, or group of users has to a resource. This includes an Amazon S3 bucket, Redshift instance, or secrets backend.
```
Comment (jwitz) [inline astro/connect-aws.md:159]:
```suggestion
To grant an Astro Deployment access to a service that is running in an AWS account not managed by Astronomer, use AWS IAM roles. IAM roles on AWS are often used to manage the level of access a specific user, object, or group of users has to a resource. This includes an Amazon S3 bucket, Redshift instance, or secrets backend.
```

**Review by jwitz — APPROVED** [review]
Thank you @melugoyal ! LGTM

File: astro/connect-aws.md (hunk 1 above)
Comment (chomipi88) [reply astro/connect-aws.md]: @melugoyal looks like we kept this line. any reason for that?

Comment (jwitz) [reply astro/connect-aws.md]: Sorry I merged before this discussion played out, happy to update this via hotfix

Comment (melugoyal) [reply astro/connect-aws.md]: no it's fine, the diff shown here is the old diff. look at the latest diff

**Pull request merged**